### ISSUE TITLE
latest_testflight_build_number is getting build number of deleted builds

### ISSUE THREAD

**Issue by yuliyan5** [fastlane/fastlane#14813]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Hello,

I am relatively new to iOS and I have been using fast lane for a couple of weeks now. Here is what I have noticed

latest_testflight_build_number is getting versions of builds which were removed from testflight. 

Currently there is only one version in Testflight (1.0), a couple of builds were deleted because of bad naming (201905....). We had 1.1.0 for a while created by mistake, which was deleted as well.
- If I don't define the version e.g version: "1.0" I get the latest build number for version 1.1.0 which is deleted
- When I define the version: "1.0", I get the build value (21905....) which was also deleted. 

I have tried to upload a new build to test flight  with the correct version e.g 1.0.6.10033 hoping that this would overwrite the value, but it didn't work. I still get latest_testflight_build_number - 201905... for v1.0

I browsed to the tab next to Testflight, which is Activity - I am able to see all the deleted versions there. But I am not sure if this should be an issue.

##### Command executed
bundle exec fastlane run latest_testflight_build_number version:"1.0"

##### Complete output when running fastlane, including the stack trace and command used
$ bundle exec fastlane run latest_testflight_build_number version:"1.0"
[✔] 🚀 
+----------------------------+---------+---------------------------------------------------+
|                                       Used plugins                                       |
+----------------------------+---------+---------------------------------------------------+
| Plugin                     | Version | Action                                            |
+----------------------------+---------+---------------------------------------------------+
| fastlane-plugin-versioning | 0.3.4   | get_version_number_from_git_branch                |
|                            |         | get_version_number_from_plist                     |
|                            |         | get_build_number_from_plist ci_build_number       |
|                            |         | increment_version_number_in_plist                 |
|                            |         | get_app_store_version_number get_info_plist_path  |
|                            |         | increment_build_number_in_plist                   |
+----------------------------+---------+---------------------------------------------------+

[15:57:47]: --------------------------------------------
[15:57:47]: --- Step: latest_testflight_build_number ---
[15:57:47]: --------------------------------------------
[15:57:47]: To not be asked about this value, you can specify it using 'username'
[15:57:47]: Your Apple ID Username: @ 
[15:57:58]: Login to App Store Connect ( @ )
[15:58:00]: Login successful
[15:58:02]: Fetching the latest build number for version 1.0
[15:58:03]: Latest upload for version 1.0 is build: 201905221935
[15:58:03]: Result: 201905221935

### Environment

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.14.5                                                        |
| Ruby                        | 2.3.7                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.20.1 (Apple Git-117)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.14.5 (18F132)                                      |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.6.5                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 10.2.1                                                         |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "" 
team_id "" 
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.123.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                     | Version | Update-Status |
| -------------------------- | ------- | ------------- |
| fastlane-plugin-versioning | 0.3.4   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                        | Version      |
| -------------------------- | ------------ |
| did_you_mean               | 1.0.0        |
| slack-notifier             | 2.3.2        |
| atomos                     | 0.1.3        |
| CFPropertyList             | 3.0.0        |
| claide                     | 1.0.2        |
| colored2                   | 3.1.2        |
| nanaimo                    | 0.2.6        |
| xcodeproj                  | 1.9.0        |
| rouge                      | 2.0.7        |
| xcpretty                   | 0.3.0        |
| terminal-notifier          | 2.0.0        |
| unicode-display_width      | 1.6.0        |
| terminal-table             | 1.8.0        |
| plist                      | 3.5.0        |
| public_suffix              | 2.0.5        |
| addressable                | 2.6.0        |
| multipart-post             | 2.0.0        |
| word_wrap                  | 1.0.0        |
| tty-cursor                 | 0.6.1        |
| tty-spinner                | 0.9.0        |
| babosa                     | 1.0.2        |
| colored                    | 1.2          |
| highline                   | 1.7.10       |
| commander-fastlane         | 4.4.6        |
| excon                      | 0.64.0       |
| faraday                    | 0.15.4       |
| unf_ext                    | 0.0.7.6      |
| unf                        | 0.1.4        |
| domain_name                | 0.5.20180417 |
| http-cookie                | 1.0.3        |
| faraday-cookie_jar         | 0.0.6        |
| fastimage                  | 2.1.5        |
| gh_inspector               | 1.1.3        |
| mini_magick                | 4.5.1        |
| multi_json                 | 1.13.1       |
| multi_xml                  | 0.6.0        |
| rubyzip                    | 1.2.2        |
| security                   | 0.1.3        |
| xcpretty-travis-formatter  | 1.0.0        |
| dotenv                     | 2.7.2        |
| bundler                    | 2.0.1        |
| faraday_middleware         | 0.13.1       |
| naturally                  | 2.2.0        |
| simctl                     | 1.6.5        |
| uber                       | 0.1.0        |
| declarative                | 0.0.10       |
| declarative-option         | 0.1.0        |
| representable              | 3.0.4        |
| retriable                  | 3.1.2        |
| mime-types-data            | 3.2019.0331  |
| mime-types                 | 3.2.2        |
| jwt                        | 2.1.0        |
| signet                     | 0.11.0       |
| memoist                    | 0.16.0       |
| os                         | 1.0.1        |
| googleauth                 | 0.6.7        |
| httpclient                 | 2.8.3        |
| google-api-client          | 0.23.9       |
| google-cloud-env           | 1.0.5        |
| google-cloud-core          | 1.3.0        |
| digest-crc                 | 0.4.1        |
| google-cloud-storage       | 1.16.0       |
| emoji_regex                | 1.0.1        |
| tty-screen                 | 0.7.0        |
| json                       | 2.2.0        |
| io-console                 | 0.4.5        |
| libxml-ruby                | 2.9.0        |
| psych                      | 2.1.0.1      |
| fastlane-plugin-versioning | 0.3.4        |
</details>

*generated on:* **2019-05-27**
</details>

**Comment by max-ott** [comment]
Hi @yuliyan5 and welcome to Fastlane! 👋 

I'm not 100% sure what the problem is and it looks like you are mixing build ID and version ID. You might want to check out my explanation over at Stackoverflow: https://stackoverflow.com/questions/56088795/app-store-bundle-invalid-error-when-uploading-build/56150210#56150210

Could you please add your full `fastfile`? From the documentation over at https://github.com/SiarheiFedartsou/fastlane-plugin-versioning I can only see that it displays you the numbers of the `plist` and not the actual values from App Store Connect.

**Comment by yuliyan5** [comment]
Hello @max-ott,

Thank you for the response. I will add some screenshots to make it a bit more clear:
Screen one of Testflight and the latest build number which I would expect:
![image](https://user-images.githubusercontent.com/41670419/58467832-0c43b080-8145-11e9-96d3-caaa8a3e75bf.png)

Screen two of Activity tab. Here I am able to see versioons which were removed from Testflight because of wrong naming:
![image](https://user-images.githubusercontent.com/41670419/58467989-5af14a80-8145-11e9-853a-e251b9d18c3c.png)

By using latest_testflight_build_number, I am expecting to get the latest build number from the Testflight tab, not the Activity tab. In my case I get value "201905221935" instead of 1.0.7.10015. I am not sure if there is a way to delete an activity even if I had more permissions. I hope this makes sense. 

Oh, and here is the Fastfile:

`default_platform(:ios)                                                                                                                                                                                                                                                                                                                      
platform :ios do                                                                                                                                                                              
  lane :beta do
    increment_build_number(
        build_number: latest_testflight_build_number + 1, 
        xcodeproj: "./projectname.xcodeproj")  
    build_app(scheme: "projectname",                                                                    
              workspace: "projectname.xcworkspace",                                                                                                                                        
              configuration: "Release")  
    upload_to_testflight(skip_submission: true,
                         username: "@.com",
                         app_identifier: "com..")                                                                                                                                                                           
  end                                                                                                                                                 
end `

Thanks

**Comment by max-ott** [comment]
No, I think it works as expected. Even though the build might no longer be available for testing, you must increment the number from the last build in the activity tab. (This is what App Store Connect uses for validation. Same goes for the versions)

The TestFlight tab is just an interface to manage the tester-facing stuff. Information there are purely "informational".

**Comment by yuliyan5** [comment]
@max-ott  Ok, thank you very much for the clarification. I will need to check if someone can remove the builds completely then. I am closing the issue..

**Comment by max-ott** [comment]
@yuliyan5 FYI: You can't delete builds on App Store Connect. Simply increment them ;-)

**Comment by yuliyan5** [comment]
@max-ott  Yes, I just found out. I have tried to infringement the last build 1.0.6 and it was successful but after doing so, I still get the 201905 build as value. I guess because of the first digit since 2 > 1. It uses last build number sorted by build name, not by date. I guess for this version I am stuck on getting the build number which is starting with 2 when I use latest_testflight_build_number
